### ISSUE TITLE
Where to mount CLI startup scripts

### ISSUE THREAD

**Issue by simonbrommers** [bitnami/charts#6963]
Im hoping it's not just me being really rather dumb, but I cannot work out where to mount .cli scripts into the pods when launching this Helm chart.  I'm trying to automatically configure aspects of logging for keycloak using the following .cli script 👍 
`embed-server --server-config=standalone-ha.xml --std-out=echo
batch

echo Configuring logging

/subsystem=logging/json-formatter=json:add(exception-output-type=formatted, pretty-print=false, meta-data={label=value})
/subsystem=logging/console-handler=CONSOLE:write-attribute(name=named-formatter, value=json)
/subsystem=logging/logger=org.keycloak.events/:add(category=org.keycloak.events,level=DEBUG)

echo Finished configuring logging

run-batch
stop-embedded-server`

and have mounted this file under /opt/bitnami/startup-scripts or /opt/jboss/startup-scripts and neither seem to be being picked up.  Can anyone shed some light on this?  I've not found anything within the Helm chart docs.

**Comment by miguelaeh** [comment]
Hi @simonbrommers ,
What helm chart are you talking about?

**Comment by simonbrommers** [comment]
My apologies, I had meant to address this question to the Bitnami Keycloak helm chart : https://github.com/bitnami/charts/tree/master/bitnami/keycloak/

**Comment by alemorcuq** [comment]
Hi, @simonbrommers.

The Bitnami image for Keycloak supports mounting init scripts to `/docker-entrypoint-initdb.d` (https://github.com/bitnami/bitnami-docker-keycloak/blob/44787a1c715560450982717e2a06708e98645254/14/debian-10/rootfs/opt/bitnami/scripts/libkeycloak.sh#L412). However, right now it only supports `.sh` scripts and not `.cli` scripts.

I'm going to create an internal task to add support for `.cli` scripts.

**Comment by phutri2005** [comment]
I'm able to execute cli script mouting volumes in docker-compose file:
`      - "./scripts/disable-theme.sh:/docker-entrypoint-initdb.d/disable-theme.sh"`
The file disable-theme.sh content is:
`debug_execute jboss-cli.sh <<EOF
embed-server --server-config=${KEYCLOAK_CONF_FILE} --std-out=discard
batch
/subsystem=keycloak-server/theme=defaults/:write-attribute(name=cacheThemes,value=false)
/subsystem=keycloak-server/theme=defaults/:write-attribute(name=cacheTemplates,value=false)
/subsystem=keycloak-server/theme=defaults/:write-attribute(name=staticMaxAge,value=-1)
run-batch
stop-embedded-server
EOF`

**Comment by alemorcuq** [comment]
Thanks for the tip information, @phutri2005! That's very useful!

**Comment by leo-onramp** [comment]
@duhow this worked for me! thanks for posting it

**Comment by Drustan** [comment]
@duhow Do you use this with the docker.io bitnami/keycloak image?

I tried to add this in the chart values, but it doesn't seem to work :

```
    initdbScripts:
      logging.sh: |
        debug_execute jboss-cli.sh <<EOF
        embed-server --server-config=${KEYCLOAK_CONF_FILE} --std-out=discard
        batch
        /subsystem=keycloak-server/spi=eventsListener:add
        /subsystem=keycloak-server/spi=eventsListener/provider=jboss-logging:add(enabled=true)
        /subsystem=keycloak-server/spi=eventsListener/provider=jboss-logging:write-attribute(name=properties.success-level,value=info)
        /subsystem=keycloak-server/spi=eventsListener/provider=jboss-logging:write-attribute(name=properties.error-level,value=warn)
        run-batch
        stop-embedded-server
        EOF
```

I didn't find the required executables in the bitnami/keycloak image :

```
keycloak@keycloak-0:/docker-entrypoint-initdb.d$ cat logging.sh 
debug_execute jboss-cli.sh <<EOF
embed-server --server-config= --std-out=discard
batch
/subsystem=logging/json-formatter=json:add(exception-output-type=formatted, pretty-print=false, meta-data={label=value})
/subsystem=logging/console-handler=CONSOLE:write-attribute(name=named-formatter, value=json)
/subsystem=logging/logger=org.keycloak.events/:add(category=org.keycloak.events,level=DEBUG)
run-batch
stop-embedded-server
EOF
keycloak@keycloak-0:/docker-entrypoint-initdb.d$ ./logging.sh 
./logging.sh: line 1: debug_execute: command not found
```

**Comment by duhow** [comment]
@Drustan that's correct, but newer Docker image versions may not have the requested program.
I haven't updated Keycloak yet, so feel free to post any solution you find.

**Comment by KEDLogic** [comment]
Is there still no way to mount .cli scripts and if not is this in progress? I'm running 21.0.2 so the debug_execute is not an option.
